SOPT-iOS/Projects/Features/BaseFeatureDependency/Sources/Coordinator/Foundation/Coordinator.swift

```diff
@@ -21,3 +21,5 @@ public protocol Coordinator: AnyObject {
     /// - Parameter style: modal 또는 Root 등의 옵션을 지정합니다.
     func start(by style: CoordinatorStartingOption)
 }
+
+public protocol Coordinator: AnyObject { }
```

Comment:
```suggestion
public protocol AnyCoordinatorObject: AnyObject { }
```

coordinator라는 프로토콜이 아무 역할도 하지않는게 어색할 수 있어서, AnyObject의 이름을 본따 AnyCoordinatorObject란 네이밍은 어떨까요.

나중에 봤을때 Object라는 명칭이 있으면 메모리를 참조하기 위해 사용하는구나 라고 이해하기 쉬울 것 같아요

Reply:
473ec34a78ecae27fa5026da39d9c6ad120f7693

+) 기존 딥링크 구조에서 Coordinatable(리팩토링 전 Coordiantor 프로토콜)을 사용 중입니다.

Coordinatable의 start 함수를 사용하지 않아 딥링크 프로토콜 타입도 `AnyCoordinatorObject`로 변경하려했으나,
코디네이터를 `coordinator as? LegacyAttendanceCoordinator else { return nil }`로 타입캐스팅 하여 사용하기 때문에 **BaseCoordinator가 채택하는 Coordinatable & AnyCoordinatorObject 타입이나 다름없다고 생각**했습니다.

따라서 typealias Coordinator = Coordinatable & AnyCoordinatorObject를 사용하고 딥링크 코드는 수정하지 않았습니다.